File: src/main/java/com/zuminX/annotations/swagger/AuthorizationScope.java
package com.zuminX.annotations.swagger;

import com.zuminX.annotations.AnnotationAttr;
import com.zuminX.annotations.AnnotationStr;
import com.zuminX.names.ClassName;
import com.zuminX.names.SwaggerAnnotation;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * 对应io.swagger.annotations.AuthorizationScope
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class AuthorizationScope extends AnnotationStr {

  @AnnotationAttr(show = true)
  private String scope;

  @Override
  public ClassName getClassName() {
    return SwaggerAnnotation.AUTHORIZATION_SCOPE;
  }
}

File: src/main/java/com/zuminX/annotations/swagger/AuthorizationScope.kt
package com.zuminX.annotations.swagger

import com.zuminX.annotations.AnnotationAttr
import com.zuminX.annotations.AnnotationStr
import com.zuminX.names.ClassName
import com.zuminX.names.annotation.AUTHORIZATION_SCOPE

/**
 * 对应io.swagger.annotations.AuthorizationScope
 */
class AuthorizationScope : AnnotationStr() {

  @field:AnnotationAttr(show = true)
  var scope: String? = null
  
  override fun getClassName(): ClassName = AUTHORIZATION_SCOPE
}